File: B/PHP/Memory.php
<?php

class Memory extends Hardware{
    private $freq = 0;
    private $memSize = 0;
    private $cuda = false;

    public function __construct($freq = 0, $memSize = 0, $cuda = false){
        $this->freq = $freq;
        $this->memSize = $memSize;
        $this->cuda = $cuda;
    }

    public function setFreq($freq){
        $this->freq = $freq;
    }

    public function getFreq(){
        return $this->freq;
    }

    public function setMemSize($memSize){
        $this->memSize = $memSize;
    }

    public function getMemSize(){
        return $this->memSize;
    }

    public function setCuda($cuda){
        $this->cuda = $cuda;
    }

    public function getcuda(){
        return $this->cuda;
    }
}

?>

File: B/C++/Memory.cpp
#include "Hardware.cpp"

class Memory : public Hardware{
    private:
        int freq;
        int memSize;
        bool cuda;
    public:
        Memory(){
            this->freq = 0;
            this->memSize = 0;
            this->cuda = false;
        }

        void setFreq(int f){
            this->freq = f;
        }

        int getFreq(){
            return this->freq;
        }

        void setMemSize(int m){
            this->memSize = m;
        }

        int getMemSize(){
            return this->memSize;
        }

        void setCuda(bool c){
            this->cuda = c;
        }

        bool getCuda(){
            return this->cuda;
        }
};